Design.md

```diff
@@ -0,0 +1,55 @@
+# Design doc for job-runner
+
+Job-runner provides a gRCP API to execute arbitrary commands on a Linux host.
+
+## Runner
+
+The jobs themselves are normal child processes of the runner. The runner configures CGroups and Linux namespaces for children to limit resource usage and isolate the them from the other processes. It also handles output recording and forwarding. All data is stored in-memory. In case of the service process termination, all jobs and their status are lost.
+
+## Communication and auth
+
+Clients communicate via gRPC with a simple protocol: One service with four endpoints. The communication is secured [RusTLS](https://github.com/ctz/rustls) which by design supports only modern, secure cipher suites. [Tonic](https://github.com/hyperium/tonic) is used to serve gRPC and almost automatically handles protocol buffers, encryption and related concerns.
+
+Authentication is implemented with mTLS. Server and client have different CA roots, which they are expected to exchange in a secure way. (Scenario-wise: the client CA is operated by the same organization that hosts job-runner). The server identifies each client with it's CommonName (CN) field of the certificate. The client CA only issues certificates with CN values that begin with `client_` to developers that are allowed to access the API. Authorization is done by checking for this property; clients with other CN prefixes are not allowed to execute any operations. Moreover, all running jobs are bound to the CN of the client calling `Start`, and only a client with that certificate is allowed to execute operations for that job.
```

Comment:
optional: using the `client_` CN prefix sounds like a form RBAC (the CN prefix encodes the "client" role)
I suggest either encoding the role(s) separately from username/client ID, or dropping them from the design.

Reply:
That's exactly what it is. I think it's an ok simplification for this, but dropping them from this is even simpler, so I'll do that.